File: nodejs/src/apis/v1/dataSourceApi.ts
import { ApiEndpointBase } from "@lib/apis/apiEndpointBase.js"
import { HttpClient } from "@lib/services/httpClient.js"
import { DataSource, DataSourceInfo } from "@lib/types/index.js"

export class DataSourceApiV1 extends ApiEndpointBase {
    constructor(client: HttpClient) {
        super(client, "/v1/datasources")
    }

    async getAll() {
        const resRaw = await this.client.get(this.basePath)
        const res = JSON.parse(resRaw) as DataSourceInfo[]
        return res
    }
    async get(id: number) {
        const url = `${this.basePath}/${id}`
        const resRaw = await this.client.get(url)
        const res = JSON.parse(resRaw) as DataSource
        return res
    }
}

File: dotnet/DigiLean.Api.Client/V1/DatasourceApi.cs
using DigiLean.Api.Model.Clients;
using DigiLean.Api.Model.V1;
using Microsoft.Extensions.Logging;

namespace DigiLean.Api.Client.V1
{
    public class DatasourceApiV1 : ApiEndpointBase
    {
        public DatasourceApiV1(HttpClient client, ILogger logger) : base(client, logger, "/v1/datasources")
        {
        }

        /// <summary>
        /// Get all datasources
        /// </summary>
        public Task<List<DataSourceInfo>> Get()
        {
            return GetResponseAndHandleError<List<DataSourceInfo>>(BasePath);
        }

        public Task<DataSource> GetDataSource(int dataSourceId)
        {
            var url = $"{BasePath}/{dataSourceId}";
            return GetResponseAndHandleError<DataSource>(url);
        }
    }
}
